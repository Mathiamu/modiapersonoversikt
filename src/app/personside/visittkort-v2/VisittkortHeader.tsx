import * as React from 'react';
import { Data as Persondata } from './PersondataDomain';

interface Props {
    persondata: Persondata;
    erApen: boolean;
    toggleApen: () => void;
}

function VisittkortHeader(props: Props) {
    return <h1>Hello header</h1>;
}

export default VisittkortHeader;
