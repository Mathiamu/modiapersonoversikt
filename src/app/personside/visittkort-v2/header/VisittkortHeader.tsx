import * as React from 'react';
import { Data as Persondata, Kjonn } from '../PersondataDomain';
import { Normaltekst, Undertittel } from 'nav-frontend-typografi';
import styled from 'styled-components/macro';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import Kvinne from '../../../../svg/Kvinne';
import Mann from '../../../../svg/Mann';
import { hentNavn } from '../utils-visittkort';
import { useRef } from 'react';

interface Props {
    persondata: Persondata;
    erApen: boolean;
    toggleApen: () => void;
}

const VisittkortHeaderDiv = styled.section`
    background-color: white;
    padding: ${theme.margin.layout};
    padding-right: 0;
    position: relative;
    display: flex;
    flex-wrap: nowrap;
    width: 100%;
    cursor: pointer;
`;

const StyledContent = styled.div`
    flex-grow: 1;
    display: flex;
    flex-flow: row wrap;
    > * {
        flex: 1 1 15rem;
    }
`;

const VenstreFelt = styled.section`
    display: flex;
`;

const HoyreFelt = styled.div`
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    text-align: right;
    box-sizing: border-box;
`;

const IkonDiv = styled.div`
    flex: 0 0 ${theme.margin.px50};
    text-align: left;
    svg {
        height: ${theme.margin.px40};
        width: auto;
    }
`;

const GrunninfoDiv = styled.section`
    flex: 1 1;
    text-align: left;
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    word-break: break-word;
    *:focus {
        ${theme.focus}
    }
    > *:first-child {
        margin-bottom: 0.2em !important;
    }
`;

const ChevronStyling = styled.div`
    padding: ${pxToRem(10)};
    display: flex;
    align-items: center;
    justify-content: center;
`;

function erMann(props: Props) {
    const kjonn = props.persondata.person.kjonn[0];
    if (!kjonn) {
        return;
    }
    return kjonn.kode === Kjonn.M;
}

function getAlder(props: Props): string | undefined {
    const fodselsdato = props.persondata.person.fodselsdato[0];
    if (!fodselsdato) {
        return;
    }
    return (new Date().getDate() - new Date(fodselsdato).getDate()).toString();
}

function VisittkortHeader(props: Props) {
    const navneLinjeRef = useRef<HTMLSpanElement>(null);

    const ikon = {
        ikon: erMann(props) ? <Mann /> : <Kvinne />
    };

    const toggleApen = () => {
        props.toggleApen();
    };

    const alder = getAlder(props);

    return (
        <VisittkortHeaderDiv role="region" aria-label="Visittkort-hode" onClick={toggleApen}>
            <StyledContent>
                <VenstreFelt>
                    <IkonDiv>{ikon.ikon}</IkonDiv>
                    <GrunninfoDiv>
                        <Undertittel tag="h1">
                            <span ref={navneLinjeRef} tabIndex={-1} /* for at focus skal funke */>
                                {hentNavn(props.persondata.person.navn[0])} ({alder})
                            </span>
                        </Undertittel>
                    </GrunninfoDiv>
                </VenstreFelt>
                <HoyreFelt>
                    <Normaltekst>Her kommer navkontor osv</Normaltekst>
                </HoyreFelt>
                <ChevronStyling>
                    <Normaltekst>Her er det noe</Normaltekst>
                </ChevronStyling>
            </StyledContent>
        </VisittkortHeaderDiv>
    );
}

export default VisittkortHeader;
