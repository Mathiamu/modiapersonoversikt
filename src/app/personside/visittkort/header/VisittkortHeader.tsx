import * as React from 'react';
import styled from 'styled-components';
import Undertittel from 'nav-frontend-typografi/lib/undertittel';
import NavKontorContainer from './NavKontorContainer';

import { erDød, Person } from '../../../../models/person/person';
import PersonStatus from './status/PersonStatus';
import EtiketterContainer from './EtiketterContainer';

const mannPath = require('../body/familie/mann.svg');
const kvinnePath = require('../body/familie/kvinne.svg');

interface VisittkortHeaderProps {
    person: Person;
}

const VisittkortHeaderDiv = styled.div`
  width: 100%;
  display: flex;
  flex-flow: row nowrap;
`;

const VenstreFelt = styled.div`
  display: flex;
  flex-grow: 1;
  margin-right: 1em;
`;

const HøyreFelt = styled.div`
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-end;
  text-align: right;
  padding-right: 1em;
  box-sizing: border-box;
`;

const IkonDiv = styled.div`
  flex: 0 0 50px;
  text-align: left;
  > img {
    width: 40px;
  }
`;

const InfoDiv = styled.div`
  flex: 1 1;
  text-align: left;
  > *:first-child {
    margin-bottom: 0.2em !important;
  };
`;

const UpperCaseForMock = styled.span`
  text-transform: uppercase;
`;

interface PersonProps {
    person: Person;
}

function Navnelinje({person}: PersonProps) {
    const alder = erDød(person.personstatus) ? 'Død' : person.alder;
    return (
        <Undertittel>
            <UpperCaseForMock>
                {person.navn.sammensatt} ({alder})
            </UpperCaseForMock>
        </Undertittel>
    );
}

function VisittkortHeader({person}: VisittkortHeaderProps) {
    const ikon = {
        path: person.kjønn === 'M' ? mannPath : kvinnePath,
        alt: person.kjønn === 'M' ? 'Mann' : 'Kvinne'
    };
    return (
        <VisittkortHeaderDiv>

            <VenstreFelt>
                <IkonDiv>
                    <img src={ikon.path} alt={ikon.alt} title={ikon.alt}/>
                </IkonDiv>
                <InfoDiv>
                    <Navnelinje person={person}/>
                    <PersonStatus person={person}/>
                </InfoDiv>
            </VenstreFelt>

            <HøyreFelt>
                <EtiketterContainer/>
                <NavKontorContainer/>
            </HøyreFelt>

        </VisittkortHeaderDiv>
    );
}

export default VisittkortHeader;
