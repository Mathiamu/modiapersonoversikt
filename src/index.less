@nodeModulesPath: './../../';
@coreModulePath: './../../';

@import '../node_modules/nav-frontend-core/less/core';
@import '../node_modules/nav-frontend-typografi-style/src/index';
@import '../node_modules/nav-frontend-knapper-style/src/index';
@import '../node_modules/nav-frontend-ekspanderbartpanel-style/src/index';
@import '../node_modules/nav-frontend-spinner-style/src/spinner-style';
@import '../node_modules/nav-frontend-alertstriper-style/src/index';

@import './components/startbilde/startbilde.less';
@import './components/visittkort/Visittkort';


body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

.personoversikt {
  height: 100vh;
  display: flex;
  flex-direction: column;
  .main-layout{
    flex-grow: 1;
  }
}
