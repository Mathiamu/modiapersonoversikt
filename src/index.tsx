import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from './app/App';
import './index.css';
import renderDecoratorHead from './decorator';
import { parseUrlForPersonIKontekst } from './utils/urlUtils';

ReactDOM.render(
    <App />,
    document.getElementById('root') as HTMLElement
);

const fodselsnummer = parseUrlForPersonIKontekst(window.location);
renderDecoratorHead(fodselsnummer);

export { default as Visittkort } from './components/StandAloneVisittkort/VisittKortStandAlone';
