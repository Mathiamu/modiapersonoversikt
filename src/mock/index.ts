import faker from 'faker/locale/nb_NO';
import navfaker from 'nav-faker';
import Cookies from 'js-cookie';
import FetchMock, { Middleware, MiddlewareUtils, MockHandler, MockRequest } from 'yet-another-fetch-mock';
import { erGyldigFødselsnummer } from 'nav-faker/dist/personidentifikator/helpers/fodselsnummer-utils';
import { apiBaseUri } from '../api/config';
import { getPerson } from './person/personMock';
import { getMockKontaktinformasjon } from './person/krrKontaktinformasjon/kontaktinformasjon-mock';
import {
    mockGeneratorMedEnhetId,
    mockGeneratorMedFodselsnummer,
    verify,
    withDelayedResponse
} from './utils/fetch-utils';
import { getMockNavKontor } from './navkontor-mock';
import { erEgenAnsatt } from './egenansatt-mock';
import { mockBaseUrls } from './baseUrls-mock';
import { getMockVeilederRoller } from './veilderRoller-mock';
import { mockRetningsnummereKodeverk } from './kodeverk/retningsnummer-mock';
import { mockTilrettelagtKommunikasjonKodeverk } from './kodeverk/tilrettelagt-kommunikasjon-kodeverk-mock';
import { mockPostnummere } from './kodeverk/postnummer-kodeverk-mock';
import { mockLandKodeverk } from './kodeverk/land-kodeverk-mock';
import { mockValutaKodeverk } from './kodeverk/valuta-kodeverk-mock';
import { mockVergemal } from './person/vergemal/vergemalMock';
import { getMockUtbetalinger } from './utbetalinger/utbetalinger-mock';
import { getMockSykepengerRespons } from './ytelse/sykepenger-mock';
import { getMockForeldrepenger } from './ytelse/foreldrepenger-mock';
import { getMockPleiepenger } from './ytelse/pleiepenger-mock';
import { mockFeatureToggle } from './featureToggle-mock';
import { getMockSaksoversikt } from './saksoversikt/saksoversikt-mock';
import { getMockOppfolging, getMockYtelserOgKontrakter } from './oppfolging-mock';
import { getDittNavVarsler, getMockVarsler } from './varsler/varsel-mock';
import { getForeslattEnhet, getMockAnsatte, getMockEnheter, getMockGsakTema } from './meldinger/oppgave-mock';
import { getMockInnloggetSaksbehandler } from './innloggetSaksbehandler-mock';
import { saker } from './journalforing/journalforing-mock';
import { mockPersonsokResponse, mockStaticPersonsokRequest } from './person/personsokMock';
import { setupWsControlAndMock } from './context-mock';
import standardTekster from './standardTeksterMock.js';
import { getSaksBehandlersEnheterMock } from './getSaksBehandlersEnheterMock';
import { saksbehandlerCookieNavnPrefix } from '../redux/session/saksbehandlersEnhetCookieUtils';
import { OppgaverBackendMock } from './mockBackend/oppgaverBackendMock';
import { setupSaksbehandlerInnstillingerMock } from './saksbehandlerinnstillinger-mock';
import { failurerateMiddleware } from './utils/failureMiddleware';
import { setupDraftMock } from './draft-mock';
import { authMock, tilgangskontrollMock } from './tilgangskontroll-mock';
import { delayed } from './utils-mock';
import { MeldingerBackendMock } from './mockBackend/meldingerBackendMock';
import { setupSFDialogMock } from './dialoger/sf-dialoger-mock';
import { setupHenvendelseDialogMock } from './dialoger/henvendelse-dialoger-mock';
import { FeatureToggles } from '../components/featureToggle/toggleIDs';
import { getAktorId } from './aktorid-mock';
import { setupPersondataMock } from './persondata/persondata';

const STATUS_OK = () => 200;
const STATUS_BAD_REQUEST = () => 400;

const oppgaveBackendMock = new OppgaverBackendMock();
const meldingerBackendMock = new MeldingerBackendMock(oppgaveBackendMock);

const harEnhetIdSomQueryParam = (req: MockRequest) => {
    const enhetQueryParam = req.queryParams.enhet;
    if (!enhetQueryParam) {
        return 'Skal ha enhetId i queryParameter';
    }
    return undefined;
};

export function randomDelay() {
    if (navfaker.random.vektetSjanse(0.05)) {
        return faker.random.number(5000);
    }
    return faker.random.number(750);
}

const fodselsNummerErGyldigStatus = (req: MockRequest) =>
    erGyldigFødselsnummer(req.pathParams.fodselsnummer) ? STATUS_OK() : STATUS_BAD_REQUEST();

function setupInnloggetSaksbehandlerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/hode/me',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockInnloggetSaksbehandler())
    );
}

function setUpSaksbehandlersEnheterMock(mock: FetchMock) {
    mock.get(apiBaseUri + '/hode/enheter', withDelayedResponse(randomDelay(), STATUS_OK, getSaksBehandlersEnheterMock));
}

function setupTilgangskontroll(mock: FetchMock) {
    mock.get(
        '/modiapersonoversikt-api/rest/tilgang/auth',
        withDelayedResponse(randomDelay(), () => 200, authMock)
    );

    mock.get(
        '/modiapersonoversikt-api/rest/tilgang/:fodselsnummer?',
        withDelayedResponse(
            randomDelay(),
            () => (Math.random() > 0.98 ? 400 : 200),
            mockGeneratorMedFodselsnummer(tilgangskontrollMock)
        )
    );
}

function setupPersonMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getPerson(fodselsnummer))
        )
    );
}

function setupAktorIdMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer/aktorid',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getAktorId(fodselsnummer))
        )
    );
}

function setupEgenAnsattMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/egenansatt/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => erEgenAnsatt(fodselsnummer))
        )
    );
}

function setupKontaktinformasjonMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer/kontaktinformasjon',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockKontaktinformasjon(fodselsnummer))
        )
    );
}

function setupSaksoversiktMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/saker/:fodselsnummer/sakstema',
        verify(
            harEnhetIdSomQueryParam,
            withDelayedResponse(
                randomDelay(),
                fodselsNummerErGyldigStatus,
                mockGeneratorMedFodselsnummer(getMockSaksoversikt)
            )
        )
    );
}

function setupUtbetalingerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/utbetaling/:fodselsnummer',
        withDelayedResponse(randomDelay(), fodselsNummerErGyldigStatus, args =>
            getMockUtbetalinger(args.pathParams.fodselsnummer, args.queryParams.startDato, args.queryParams.sluttDato)
        )
    );
}

function setupSykepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/sykepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockSykepengerRespons(fodselsnummer))
        )
    );
}

function setupForeldrepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/foreldrepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockForeldrepenger(fodselsnummer))
        )
    );
}

function setupPleiepengerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/ytelse/pleiepenger/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockPleiepenger(fodselsnummer))
        )
    );
}

function setupOppfolgingMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppfolging/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockOppfolging(fodselsnummer))
        )
    );
}

function setupYtelserOgKontrakter(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppfolging/:fodselsnummer/ytelserogkontrakter',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockYtelserOgKontrakter(fodselsnummer))
        )
    );
}

function setupVarselMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/varsler/:fodselsnummer',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => getMockVarsler(fodselsnummer))
        )
    );

    const dittnaveventHandler: MockHandler = (req, res, ctx) => {
        const headers: any = req.init?.headers;
        const fnr = headers.fodselsnummer;
        if (!erGyldigFødselsnummer(fnr)) {
            return res(ctx.status(400));
        }
        return res(ctx.status(200), ctx.json(getDittNavVarsler(fnr)));
    };

    mock.get(
        '/modiapersonoversikt/proxy/dittnav-eventer-modia/fetch/beskjed/all',
        delayed(randomDelay(), dittnaveventHandler)
    );
    mock.get(
        '/modiapersonoversikt/proxy/dittnav-eventer-modia/fetch/oppgave/all',
        delayed(randomDelay(), dittnaveventHandler)
    );
}

function setupGsakTemaMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/dialogoppgave/tema',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockGsakTema())
    );
}

function setupOppgaveEnhetMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/oppgavebehandlere/alle',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockEnheter())
    );
}

function setupForeslatteEnheterMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/oppgavebehandlere/foreslatte',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getForeslattEnhet())
    );
}

function setupAnsattePaaEnhetMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter/:enhetId/ansatte',
        withDelayedResponse(
            randomDelay(),
            STATUS_OK,
            mockGeneratorMedEnhetId(enhetId => getMockAnsatte(enhetId))
        )
    );
}

function setupGeografiskTilknytningMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/enheter',
        withDelayedResponse(
            randomDelay(),
            (args: MockRequest) => {
                if (isNaN(args.queryParams.gt) && !args.queryParams.dkode) {
                    return 404;
                } else {
                    return 200;
                }
            },
            (args: MockRequest) => getMockNavKontor(args.queryParams.gt, args.queryParams.dkode)
        )
    );
}

function setupPersonsokMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/personsok',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockPersonsokResponse(mockStaticPersonsokRequest()))
    );
}

function setSaksbehandlerinnstillingerMockBackend(args: MockRequest) {
    Cookies.set(saksbehandlerCookieNavnPrefix + '-Z990099', args.body);
    return args.body;
}

function setupVelgEnhetMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/hode/velgenhet',
        withDelayedResponse(randomDelay(), STATUS_OK, setSaksbehandlerinnstillingerMockBackend)
    );
}

function setupOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/oppgaver/plukk/:temagruppe',
        verify(
            harEnhetIdSomQueryParam,
            withDelayedResponse(randomDelay(), STATUS_OK, () => oppgaveBackendMock.plukkOppgave())
        )
    );
}

function setupTildelteOppgaverMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/oppgaver/tildelt/:fnr',
        withDelayedResponse(randomDelay(), STATUS_OK, () => oppgaveBackendMock.getTildelteOppgaver())
    );
}

function setupLeggTilbakeOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/oppgaver/legg-tilbake',
        withDelayedResponse(randomDelay(), STATUS_OK, request => oppgaveBackendMock.leggTilbake(request.body))
    );
}

function setupVergemalMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/person/:fodselsnummer/vergemal',
        withDelayedResponse(
            randomDelay(),
            fodselsNummerErGyldigStatus,
            mockGeneratorMedFodselsnummer(fodselsnummer => mockVergemal(fodselsnummer))
        )
    );
}

function setupBaseUrlsMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/baseurls',
        withDelayedResponse(randomDelay(), STATUS_OK, () => {
            return mockBaseUrls();
        })
    );
}

function setupFeatureToggleMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/featuretoggle/:toggleId',
        withDelayedResponse(
            randomDelay(),
            STATUS_OK,
            // @ts-ignore
            (args: HandlerArgument) => mockFeatureToggle(args.pathParams.toggleId)
        )
    );
}

function setupVeilederRollerMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/veileder/roller',
        withDelayedResponse(randomDelay(), STATUS_OK, () => getMockVeilederRoller())
    );
}

function setupRetningsnummerKodeverkMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Retningsnumre',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockRetningsnummereKodeverk())
    );
}

function setupTilrettelagtKommunikasjonKodeverkMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/TilrettelagtKommunikasjon',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockTilrettelagtKommunikasjonKodeverk())
    );
}

function setupPostnummerKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Postnummer',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockPostnummere())
    );
}

function setupLandKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Landkoder',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockLandKodeverk())
    );
}

function setupValutaKodeverk(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/kodeverk/Valutaer',
        withDelayedResponse(randomDelay(), STATUS_OK, () => mockValutaKodeverk())
    );
}

function setupJournalforingMock(mock: FetchMock) {
    mock.get(
        apiBaseUri + '/journalforing/:fnr/saker/',
        withDelayedResponse(randomDelay(), STATUS_OK, () => saker)
    );
    mock.post(
        apiBaseUri + '/journalforing/:fnr/:traadId',
        verify(
            harEnhetIdSomQueryParam,
            withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
        )
    );
}

function opprettOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogoppgave/opprett',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function opprettSkjermetOppgaveMock(mock: FetchMock) {
    mock.post(
        apiBaseUri + '/dialogoppgave/opprettskjermetoppgave',
        withDelayedResponse(randomDelay(), STATUS_OK, () => ({}))
    );
}

function setupStandardteksterMock(mock: FetchMock) {
    mock.get(
        '/modiapersonoversikt-skrivestotte/skrivestotte',
        withDelayedResponse(randomDelay(), STATUS_OK, () => standardTekster)
    );

    mock.post(
        '/modiapersonoversikt-skrivestotte/skrivestotte/statistikk/:id',
        withDelayedResponse(randomDelay(), STATUS_OK, () => undefined)
    );
}

const contentTypeMiddleware: Middleware = (requestArgs, response) => {
    if (response.headers) {
        return response;
    }
    response.headers = {
        'content-type': 'application/json'
    };
    return response;
};

console.log('=========================='); // tslint:disable-line
console.log('======== MED MOCK ========'); // tslint:disable-line
console.log('=========================='); // tslint:disable-line

const mock = FetchMock.configure({
    enableFallback: true,
    middleware: MiddlewareUtils.combine(
        contentTypeMiddleware,
        failurerateMiddleware(0.02),
        MiddlewareUtils.loggingMiddleware()
    )
});

setupInnloggetSaksbehandlerMock(mock);
setupPersonMock(mock);
setupPersondataMock(mock);
setupTilgangskontroll(mock);
setupEgenAnsattMock(mock);
setupKontaktinformasjonMock(mock);
setupGeografiskTilknytningMock(mock);
setupSaksoversiktMock(mock);
setupUtbetalingerMock(mock);
setupSykepengerMock(mock);
setupForeldrepengerMock(mock);
setupPleiepengerMock(mock);
setupOppgaveMock(mock);
setupAktorIdMock(mock);

if (mockFeatureToggle(FeatureToggles.BrukSalesforceDialoger)) {
    setupSFDialogMock(mock, meldingerBackendMock);
} else {
    setupHenvendelseDialogMock(mock, meldingerBackendMock);
}

setupTildelteOppgaverMock(mock);
setupLeggTilbakeOppgaveMock(mock);
setupVergemalMock(mock);
setupBaseUrlsMock(mock);
setupFeatureToggleMock(mock);
setupVeilederRollerMock(mock);
setupRetningsnummerKodeverkMock(mock);
setupTilrettelagtKommunikasjonKodeverkMock(mock);
setupPostnummerKodeverk(mock);
setupWsControlAndMock(mock);
setupLandKodeverk(mock);
setupValutaKodeverk(mock);
setupOppfolgingMock(mock);
setupGsakTemaMock(mock);
setupOppgaveEnhetMock(mock);
setupForeslatteEnheterMock(mock);
setupAnsattePaaEnhetMock(mock);
setupYtelserOgKontrakter(mock);
setupVarselMock(mock);
opprettOppgaveMock(mock);
opprettSkjermetOppgaveMock(mock);
setupPersonsokMock(mock);
setupJournalforingMock(mock);
setupStandardteksterMock(mock);
setupVelgEnhetMock(mock);
setUpSaksbehandlersEnheterMock(mock);
setupSaksbehandlerInnstillingerMock(mock);
setupDraftMock(mock);
