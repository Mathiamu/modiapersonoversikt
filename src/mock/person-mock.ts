import * as faker from 'faker/locale/nb_NO';

import { Bankkonto, Person } from '../models/person';
import { Diskresjonskoder } from '../constants';
import { vektetSjanse } from './utils';
import * as moment from 'moment';

function erMann(fødselsnummer: string) {
    return Number(fødselsnummer.charAt(8)) % 2 === 1;
}

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: 'M',
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE
};

export const bankkontoNorsk: Bankkonto = {
    erNorskKonto: true,
    bank: 'Nordea ASA',
    kontonummer: Number(faker.finance.account(11)),
    sistEndretAv: '1010800 BD03',
    sistEndret: getSistOppdatert(),
};

export const bankkontoUtland: Bankkonto = {
    erNorskKonto: false,
    bank: 'BBVA',
    kontonummer: Number(faker.finance.account(9)),
    sistEndretAv: '1010800 BD03',
    sistEndret: getSistOppdatert(),
};

export function getPerson(fødselsnummer: string): Person {
    if (fødselsnummer === aremark.fødselsnummer) {
        return aremark;
    } else {
        faker.seed(Number(fødselsnummer));
        return getTilfeldigPerson(fødselsnummer);
    }
}

function getTilfeldigPerson(fødselsnummer: string): Person {
    const fornavn = getFornavn(fødselsnummer);
    const etternavn = faker.name.lastName();
    const mellomnavn = '';
    return {
        fødselsnummer: fødselsnummer,
        kjønn: erMann(fødselsnummer) ? 'M' : 'K',
        geografiskTilknytning: getGeografiskTilknytning(),
        alder: faker.random.number(100),
        navn: {
            fornavn: fornavn,
            etternavn: etternavn,
            mellomnavn: mellomnavn,
            sammensatt: `${fornavn} ${mellomnavn} ${etternavn}`
        },
        diskresjonskode: getDiskresjonskode(),
        bankkonto: getBankKonto()
    };
}

function getBankKonto(): Bankkonto | undefined {
    if (vektetSjanse(faker, 0.7)) {
        return bankkontoNorsk;
    } else if (vektetSjanse(faker, 0.2)) {
        return bankkontoUtland;
    } else {
        return undefined;
    }
}

function getDiskresjonskode() {
    if (vektetSjanse(faker, 0.1)) {
        return Diskresjonskoder.FORTROLIG_ADRESSE;
    } else if (vektetSjanse(faker, 0.1)) {
        return Diskresjonskoder.STRENGT_FORTROLIG_ADRESSE;
    } else {
        return undefined;
    }
}

function getFornavn(fødselsnummer: string): string {
    if (Number(fødselsnummer.charAt(8)) % 2 === 0 ) {
        return faker.name.firstName(1);
    } else {
        return faker.name.firstName(0);
    }
}

function getGeografiskTilknytning() {
    if (vektetSjanse(faker, 0.8)) {
        return String(faker.random.number(9999));
    } else {
        return undefined;
    }
}

function getSistOppdatert() {
    return moment(faker.date.past(5)).format(moment.ISO_8601.__momentBuiltinFormatBrand);
}