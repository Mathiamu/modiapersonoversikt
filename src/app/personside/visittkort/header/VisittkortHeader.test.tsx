import '../../../../../src/testSetup';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { aremark } from '../../../../mock/person-mock';
import VisittkortHeader from './VisittkortHeader';

test.skip('viser info om bruker i visittkort-header', () => {
    const visittkortheader = renderer.create(
        <VisittkortHeader person={aremark} />
    );

    const json = visittkortheader.toJSON();
    expect(json).toMatchSnapshot();
});
