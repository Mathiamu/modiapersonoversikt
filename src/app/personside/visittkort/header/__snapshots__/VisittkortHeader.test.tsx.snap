// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser info om bruker i visittkort-header 1`] = `
<div
  className="sc-ifAKCX hMtwKY"
>
  <div
    className="sc-EHOje UXQEB"
  >
    <div
      className="sc-bZQynM gSaPyk"
    >
      <img
        alt="Mann"
        src="mann.svg"
      />
    </div>
    <div
      className="sc-gzVnrw cfMaXP"
    >
      <h2
        className="typo-undertittel"
      >
        AREMARK TESTFAMILIEN
         (
        42
        )
      </h2>
      <p
        className="typo-undertekst"
      >
        <span
          className="sc-bxivhb jNgUAM"
        >
          10108000398
           
          <span
            className="sc-htpNat gVwGYF"
          >
            NORSK
             / 
            Gift
             / 
            Ingen barn under 21 Ã¥r 
          </span>
        </span>
      </p>
    </div>
  </div>
  <div
    className="sc-htoDjs cTcjGg"
  >
    <div
      className="sc-bwzfXH eCbFXJ"
    >
      <div
        className="etikett etikett--advarsel"
      >
        <span
          className="typo-normal"
        >
          Fortrolig adresse
        </span>
      </div>
    </div>
    <p
      className="typo-undertekst"
    >
      <span
        className="sc-bdVaJa ktDPVU"
      >
        Nav-enhet / 
        0118
         
        Nav Aremark
      </span>
    </p>
  </div>
</div>
`;
